src/relay.rs

```diff
@@ -180,6 +180,7 @@ impl<T: DuplexTransport + 'static> Network<T> {
             keydir: keydir.to_string(),
             password: password.to_string(),
             nonce,
+            pending: RwLock::new(LruCache::new(1024)),
```

Comment:
The goal is to prevent the map from growing forever. Since relay is ostensibly a long running process, we don't want that to grow until we oom. 

But yes, it does seem to be a way to dox us. At the very least it could cause an unapprove to not be triggered if an approved transfer is dropped from pending, and then removed from the chain.

Reply:
Ah, we can of course use a combination of the LRU cache and reading from the blockchain to determine what to do if  the entry does not exist.

Reply:
Then the problem is how to handle the case where a log is removed, but in that 1 second timeout before the process log future checks the cache, it is dropped. 
We would lose the state and send the approval without realizing it was removed.